TestVectorsAwsCryptographicMaterialProviders/dafny/TestVectorsAwsCryptographicMaterialProviders/test/fuzz_generator.py

```diff
@@ -0,0 +1,370 @@
+        key_desc = test_vector.get("encryptKeyDescription", {})
+        key_type = key_desc.get("type")
+        
+        # Handle both "raw" (AES) and "raw-ecdh" (ECC) types
+        if key_type in ["raw", "raw-ecdh"]:
+            # The lookup key is now the fuzzed_key_id (stored in "key" field)
+            fuzzed_key_id = key_desc["key"]
+            
+            # Get the key-id that should go in the material (different from lookup key)
+            key_id_in_material = key_desc.get(INTERNAL_KEY_ID_FIELD, fuzzed_key_id)
+            
+            if key_type == "raw":
+                encryption_algorithm = key_desc.get("encryption-algorithm", "aes")
+                base_key_id = "aes-256"  # default fallback
+                for key_type_name in RAW_KEY_TYPES:
+                    if key_type_name.startswith(encryption_algorithm):
+                        base_key_id = key_type_name
+                        break
+            else:  # raw-ecdh
+                # For ECC, use the curve name directly
+                base_key_id = key_desc.get("ecc-curve", "ecc-256")
+            
+            key_info = KEY_MATERIALS.get(base_key_id, KEY_MATERIALS["aes-256"])
+            
+            # Create the key entry for keys.json based on algorithm type
+            if base_key_id.startswith("aes"):
+                new_keys[fuzzed_key_id] = {
+                    "encrypt": True, 
+                    "decrypt": True, 
+                    "algorithm": "aes", 
+                    "type": "symmetric",
+                    "bits": key_info["bits"], 
+                    "encoding": "base64",
+                    "material": key_info["material"], 
+                    "key-id": key_id_in_material
+                }
+            elif base_key_id.startswith("ecc"):
+                new_keys[fuzzed_key_id] = {
+                    "encrypt": True,
+                    "decrypt": True,
+                    "algorithm": "ecdh",
+                    "type": "ecc-private",
+                    "bits": key_info["bits"],
+                    "encoding": "pem",
+                    "sender-material": key_info["sender-material"],
+                    "recipient-material": key_info["recipient-material"],
+                    "public-key-encoding": "base64-der",
+                    "sender-material-public-key": key_info["sender-public-key"],
+                    "recipient-material-public-key": key_info["recipient-public-key"],
+                    "key-id": key_id_in_material
+                }
+    
+    return new_keys
+
+def generate_fuzz_test_vectors(num_vectors) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    """Generate multiple fuzzed test vectors and collect new key generated."""
+    test_vectors = {}
+    
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=NonInteractiveExampleWarning)
+        for i in range(num_vectors):
+            # TODO: remove .example() usage.
```

Comment:
This TODO is non blocking because we are only using to generate test vectors.

.example() turns Hypothesis into “just a fancy random data generator,” losing the key feature of systematically exploring the input space. We are no longer getting its shrinking, edge case finding, etc but today we don't use for that purpose. However, thinking of long term this is the first thing I would like to get done because "today" it will be easier to remove .example() but if we try to remove it years after "today" it might get very hard as I expect we will add new things in here.